evm/src/Transceiver/WormholeTransceiver/WormholeTransceiverState.sol

```diff
@@ -115,233 +146,73 @@ contract WormholeTransceiver is Transceiver, IWormholeTransceiver, IWormholeRece
-
-        // parse the encoded Transceiver payload
-        TransceiverStructs.TransceiverMessage memory parsedTransceiverMessage;
-        TransceiverStructs.NttManagerMessage memory parsedNttManagerMessage;
-        (parsedTransceiverMessage, parsedNttManagerMessage) = TransceiverStructs
-            .parseTransceiverAndNttManagerMessage(WH_TRANSCEIVER_PAYLOAD_PREFIX, payload);
-
-        _deliverToNttManager(
-            sourceChain,
-            parsedTransceiverMessage.sourceNttManagerAddress,
-            parsedTransceiverMessage.recipientNttManagerAddress,
-            parsedNttManagerMessage
-        );
+    /// @inheritdoc IWormholeTransceiverState
+    function setIsWormholeEvmChain(uint16 chainId) external onlyOwner {
+        _setIsWormholeEvmChain(chainId);
     }
 
-    /// @notice Receive an attested message from the verification layer
-    ///         This function should verify the encodedVm and then deliver the attestation to the transceiver nttManager contract.
-    function receiveMessage(bytes memory encodedMessage) external {
-        uint16 sourceChainId;
-        bytes memory payload;
-        (sourceChainId, payload) = _verifyMessage(encodedMessage);
-
-        // parse the encoded Transceiver payload
-        TransceiverStructs.TransceiverMessage memory parsedTransceiverMessage;
-        TransceiverStructs.NttManagerMessage memory parsedNttManagerMessage;
-        (parsedTransceiverMessage, parsedNttManagerMessage) = TransceiverStructs
-            .parseTransceiverAndNttManagerMessage(WH_TRANSCEIVER_PAYLOAD_PREFIX, payload);
-
-        _deliverToNttManager(
-            sourceChainId,
-            parsedTransceiverMessage.sourceNttManagerAddress,
-            parsedTransceiverMessage.recipientNttManagerAddress,
-            parsedNttManagerMessage
-        );
+    /// @inheritdoc IWormholeTransceiverState
+    function setIsWormholeRelayingEnabled(uint16 chainId, bool isEnabled) external onlyOwner {
+        _setIsWormholeRelayingEnabled(chainId, isEnabled);
     }
 
-    function _verifyMessage(bytes memory encodedMessage) internal returns (uint16, bytes memory) {
-        // verify VAA against Wormhole Core Bridge contract
-        (IWormhole.VM memory vm, bool valid, string memory reason) =
-            wormhole.parseAndVerifyVM(encodedMessage);
-
-        // ensure that the VAA is valid
-        if (!valid) {
-            revert InvalidVaa(reason);
-        }
-
-        // ensure that the message came from a registered peer contract
-        if (!_verifyBridgeVM(vm)) {
-            revert InvalidWormholePeer(vm.emitterChainId, vm.emitterAddress);
-        }
+    // ============= Internal ===============================================================
 
-        // save the VAA hash in storage to protect against replay attacks.
-        if (isVAAConsumed(vm.hash)) {
-            revert TransferAlreadyCompleted(vm.hash);
+    function _setIsWormholeEvmChain(uint16 chainId) internal {
```

Comment:
Many of these internal methods are only called from the admin methods above. Rather than having this layer of indirection can we include them directly in the public admin method? Saves us some bytecode

Reply:
Yes, definitely.